### PR TITLE
Allow multiple tidal energy farms per subdomain

### PR DESCRIPTION
**Pull request by cpjordan**
- Farms must now be defined in a list for each subdomain ID
- Warning provided when a list is not used
- Corresponding updates to examples

### CODE REVIEW

File: thetis/options.py [hunk 1]
```diff
@@ -881,12 +881,12 @@ class ModelOptions2d(CommonModelOptions):
 
         Note this is only relevant if `use_automatic_wetting_and_drying_alpha` is set to ``True``.
         """).tag(config=True)
-    tidal_turbine_farms = Dict(trait=TidalTurbineFarmOptions(),
-                               default_value={}, help='Dictionary mapping subdomain ids to the options of the corresponding farm')
+    tidal_turbine_farms = Dict(trait=List(TidalTurbineFarmOptions()),
+                               default_value={}, help='Dictionary mapping subdomain ids to the options of the corresponding farm(s)')
```
Comment (stephankramer) [inline thetis/options.py:885]:
```suggestion
                               default_value={}, help='Dictionary mapping subdomain ids to a list of TidalTurbineFarmOptions instances corresponding
                               to one or more farms.')
```

**Review by stephankramer — APPROVED** [review]
Nice. Looks good. Just a small suggestion and one question: what is the use case for having multiple farms per domain id for the continuous case? Presumably that means having multiple turbine densities that act in the same area? I'm fine with also changing that to a list but was just wondering

File: thetis/options.py (hunk 1 above)
Comment (thangel) [reply thetis/options.py:885]: I suppose you could have different densities for different turbine characteristics?

Comment (cpjordan) [reply thetis/options.py:885]: I've just done that out of consistency more than anything, happy to go with whatever is preferred!

**Review by thangel — APPROVED** [review]

**Pull request merged**